test/viewmodels/users_ranking_view_model_test.dart

```diff
@@ -0,0 +1,134 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:fake_cloud_firestore/fake_cloud_firestore.dart";
+import "package:flutter/material.dart";
+import "package:flutter_test/flutter_test.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/user/user_data.dart";
+import "package:proxima/models/database/user/user_firestore.dart";
+import "package:proxima/models/database/user/user_id_firestore.dart";
+import "package:proxima/viewmodels/users_ranking_view_model.dart";
+
+import "../mocks/data/firestore_user.dart";
+import "../mocks/providers/provider_ranking.dart";
+
+void main() {
+  group("User ranking view model testing", () {
+    late FakeFirebaseFirestore fakeFirestore;
+    late ProviderContainer rankingContainer;
+
+    setUp(() async {
+      fakeFirestore = FakeFirebaseFirestore();
+      await FirestoreUserGenerator.addUsers(
+        fakeFirestore,
+        UsersRankingViewModel.rankingLimit * 2,
+      );
+
+      rankingContainer = await rankingProviderContainer(fakeFirestore);
+    });
+
+    test("Correct number of users returned", () async {
+      final ranking =
+          await rankingContainer.read(usersRankingViewModelProvider.future);
+
+      expect(
+        ranking.rankElementDetailsList.length,
+        UsersRankingViewModel.rankingLimit,
+      );
+    });
+
+    test("Logged in user has correct values", () async {
+      final ranking =
+          await rankingContainer.read(usersRankingViewModelProvider.future);
+      final currentUserRanking = ranking.userRankElementDetails;
+
+      expect(
+        currentUserRanking.userDisplayName,
+        testingUserData.displayName,
+      );
+      expect(
+        currentUserRanking.userUserName,
+        testingUserData.username,
+      );
+      expect(
+        currentUserRanking.centauriPoints,
+        testingUserData.centauriPoints,
+      );
+      expect(
+        currentUserRanking.userRank,
+        isNull,
+      );
+    });
+
+    test("View model returns a descending order with correct ranks", () async {
+      final ranking =
+          await rankingContainer.read(usersRankingViewModelProvider.future);
+      final leaderboard = ranking.rankElementDetailsList;
+
+      // Check that all ranks are non null and correct
+      final lastRank =
+          leaderboard.map((e) => e.userRank!).reduce((value, element) {
+        expect(value + 1, element);
+        return element;
+      });
+      expect(lastRank, UsersRankingViewModel.rankingLimit);
```

Comment:
Your version does two iterations (over 200 items) when it's really not necessary. 
Maybe it would be interesting to note what was complicated to understand ?
I am just using `reduce` which is a core fundamental functional programming concept.
I find it very elegant the way it is written here, usage of an iterator to check the previous value is the next one minus 1.
I will add comments if you think it is important.

(nitpick)

Reply:
After conducting a poll on our team group chat, we'll go with a third (equivalent) option:

```dart
for (final i = 0; i < rankingLimit; ++i) {
    expect(actualRank[i], i + i);
}
```

Reply:
Applied in a591228.